sdk/extensions/chronopost/tests/chronopost/shipment.py

```diff
@@ -0,0 +1,255 @@
+                f"{gateway.settings.server_url}/shipping-cxf/ShippingServiceWS",
+            )
+
+    def test_cancel_shipment(self):
+        with patch("karrio.mappers.chronopost.proxy.lib.request") as mock:
+            mock.return_value = "<a></a>"
+            karrio.Shipment.cancel(self.ShipmentCancelRequest).from_(gateway)
+
+            self.assertEqual(
+                mock.call_args[1]["url"],
+                f"{gateway.settings.server_url}/tracking-cxf/TrackingServiceWS",
+            )
+
+    def test_parse_shipment_response(self):
+        with patch("karrio.mappers.chronopost.proxy.lib.request") as mock:
+            mock.return_value = ShipmentResponse
+            parsed_response = (
+                karrio.Shipment.create(self.ShipmentRequest).from_(gateway).parse()
+            )
+            self.assertListEqual(lib.to_dict(parsed_response), ParsedShipmentResponse)
+
+    def test_parse_cancel_shipment_error_response(self):
+        with patch("karrio.mappers.chronopost.proxy.lib.request") as mock:
+            mock.return_value = ""
+            parsed_response = (
+                karrio.Shipment.cancel(self.ShipmentCancelRequest)
+                .from_(gateway)
+                .parse()
+            )
+            print(lib.to_dict(parsed_response))
+            self.assertListEqual(
+                lib.to_dict(parsed_response), ParsedCancelShipmentResponse
+            )
+
+
+if __name__ == "__main__":
+    unittest.main()
+
+
+ShipmentPayload = {
+    "service": "chronopost_13",
+    "shipper": {
+        "company_name": "Chef Royale",
+        "person_name": "Jean Dupont",
+        "address_line1": "28 rue du Clair Bocage",
+        "city": "La Seyne-sur-mer",
+        "postal_code": "83500",
+        "country_code": "FR",
+        "phone_number": "+330447110494",
+    },
+    "recipient": {
+        "company_name": "HautSide",
+        "person_name": "Lucas Dupont",
+        "address_line1": "72 rue Reine Elisabeth",
+        "city": "Menton",
+        "postal_code": "06500",
+        "country_code": "FR",
+    },
+    "parcels": [
+        {
+            "height": 15,
+            "length": 60.0,
+            "width": 30,
+            "weight": 5.0,
+            "weight_unit": "KG",
+            "dimension_unit": "CM",
+        }
+    ],
+    "label_type": "PDF",
+    "reference": "Ref. 123456",
+    "options": {"shipment_date": "2022-08-16T22:56:11"},
+}
+
+ShipmentCancelPayload = {
+    "shipment_identifier": "794947717776",
+}
+
+ParsedShipmentResponse = [
+    {
+        "carrier_id": "chronopost",
+        "carrier_name": "chronopost",
+        "docs": {},
+        "meta": {},
+    },
+    [],
+]
+
+ParsedCancelShipmentResponse = [
+    None,
+    [
+        {
+            "carrier_id": "chronopost",
+            "carrier_name": "chronopost",
+            "code": 4,
+            "message": "Document is empty, line 1, column 1 (<string>, line 1)",
+        }
+    ],
+]
```

Comment:
And that one too.

Reply:
Can you verify the commits again? I think that you commented on an outdated commit 🤔

Reply:
my bad. Thanks